cedar-policy-core/src/entities.rs

```diff
@@ -2263,7 +2263,7 @@ mod schema_based_parsing_tests {
             .from_json_value(entitiesjson)
             .expect_err("should fail due to type mismatch on hr_contacts");
         assert!(
-            err.to_string().contains(r#"in attribute `hr_contacts` on `Employee::"12UA45"`, type mismatch: attribute was expected to have type (set of (entity of type `HR`)), but actually has type record"#),
+            err.to_string().contains(r#"in attribute `hr_contacts` on `Employee::"12UA45"`, type mismatch: attribute was expected to have type (set of (entity of type `HR`)), but it does not: `{"id": "aaaaa", "type": "HR"}`"#),
```

Comment:
I think this error message is significantly worse than the old one.

Reply:
On the whole do you think the new error messages are on average worse than the old ones?  I agree this one is worse; I think some others are improved.  But if the change is on average for the negative, I can revert and just make an uglier message in the Unknown case